agent/app/agent.go

```diff
@@ -395,7 +395,11 @@ func (agent *ecsAgent) doStart(containerChangeEventStream *eventstream.EventStre
 		}
 		return exitcodes.ExitTerminal
 	}
-	agent.serviceconnectManager.SetECSClient(client, agent.containerInstanceARN)
+	scManager := agent.serviceconnectManager
+	scManager.SetECSClient(client, agent.containerInstanceARN)
+	if loaded, _ := scManager.IsLoaded(agent.dockerClient); loaded {
+		imageManager.AddImageToCleanUpExclusionList(agent.serviceconnectManager.GetLoadedImageName())
```

Comment:
Could we add the appnet agent image to agent.cfg and then exclude it like other images from the method buildImageCleanupExclusionList ?

Reply:
There are some dependency issues here
- `buildImageCleanupExclusionList` is called when ImageManager gets initiated, before `doStart()`
- SC container image is loaded when we make RegisterContainerInstance call (when we collect capabilities), during `doStart()`

We may be able to move things around in the bootstrap, but we'd need to make sure doing that does not have side effects, at which point I feel like adding a handler for excluding new images would be easier.

Again, the other images are all "known" and have fixed names and tags, which is not the case for SC container image.

Reply:
Agreed, with this approach we don't to manually keep the ImageCleanupExclusionList up to date whenever there is a new appnet agent interface version.